moveit_ros/planning/rdf_loader/src/rdf_loader.cpp

```diff
@@ -62,68 +62,51 @@ RDFLoader::RDFLoader(const std::shared_ptr<rclcpp::Node>& node, const std::strin
   moveit::tools::Profiler::ScopedStart prof_start;
   moveit::tools::Profiler::ScopedBlock prof_block("RDFLoader(robot_description)");
 
-  auto start = std::chrono::system_clock::now();
+  auto start = node->now();
 
-  std::string content;
-  // TODO(JafarAbdi): Revise parameter lookup
-  if (node->has_parameter(robot_description))
-  {
-    rclcpp::Parameter robot_description_param = node->get_parameter(robot_description);
-    try
-    {
-      content = robot_description_param.as_string();
-    }
-    catch (const rclcpp::ParameterTypeException& e)
-    {
-      RCLCPP_WARN(LOGGER, "When getting robot_description parameter %s", e.what());
-    }
-  }
-  if (content.empty())
+  // Check if the robot_description parameter is declared, declare it if it's not declared yet
+  if (!node->has_parameter(robot_description))
+    node->declare_parameter(robot_description);
+  std::string robot_description_content;
+  node->get_parameter_or(robot_description, robot_description_content, std::string());
+
+  if (robot_description_content.empty())
   {
     RCLCPP_INFO_ONCE(LOGGER, "Robot model parameter not found! Did you remap '%s'?\n", robot_description.c_str());
     return;
   }
 
   urdf::Model* umodel = new urdf::Model();
-  if (!umodel->initString(content))
+  if (!umodel->initString(robot_description_content))
   {
-    RCLCPP_INFO(LOGGER, "Unable to parse URDF from parameter: '%s'", robot_description_.c_str());
```

Comment:
I don't see the `robot_description_` parameter is used anymore, should I remove it .?

Reply:
Yes, makes sense